src/reverse.zig

```diff
@@ -0,0 +1,151 @@
+const std = @import("std");
+const address = @import("./address.zig");
+const dns = @import("./lib.zig");
+const assert = std.debug.assert;
+
+/// ReverseLookup is the primary interface for classless IN-ADDR.ARPA (IPv4) and IP6.ARPA (IPv6) delegations
+pub const ReverseLookup = struct {
+    allocator: std.mem.Allocator,
+    ip_address: []const u8,
+    packet_id: u16, // Arbitrary packet ID
+
+    const Self = @This();
+    const max_ipv6_label_size = 35;
+    const max_ipv4_label_size = 6;
+
+    pub fn init(allocator: std.mem.Allocator, ip_address: []const u8, packet_id: u16) !Self {
+        return Self{
+            .allocator = allocator,
+            .ip_address = ip_address,
+            .packet_id = packet_id,
+        };
+    }
+
+    /// Reverse lookup on a given ipv4 ip address
+    pub fn lookupIpv4(self: Self) ![][]const u8 {
+        var add = try address.IpAddress.init(self.allocator, try address.AddressMeta.Ipv4().fromString(self.ip_address));
+        const arpa_address = try add.reverseIpv4();
+
+        var labels: [max_ipv4_label_size][]const u8 = undefined;
+        const apra_address_dns_name = try dns.Name.fromString(arpa_address, &labels);
+
+        return try self.buildAndSendPacket(apra_address_dns_name);
+    }
+
+    /// Reverse lookup on a given ipv6 ip address
+    pub fn lookupIpv6(self: Self) ![][]const u8 {
+        var add = try address.IpAddress.init(self.allocator, try address.AddressMeta.Ipv6().fromString(self.ip_address));
+        const arpa_address = try add.reverseIpv6();
+
+        var labels: [max_ipv6_label_size][]const u8 = undefined;
+        const apra_address_dns_name = try dns.Name.fromString(arpa_address, &labels);
+
+        return self.buildAndSendPacket(apra_address_dns_name) catch &[_][]const u8{};
+    }
+
+    /// Internal function to build and send the DNS packet for reverse lookup agnostic of IP address type (ipv4 | ipv6)
+    fn buildAndSendPacket(self: Self, apra_address_dns_name: dns.Name) ![][]const u8 {
+        var name_pool = dns.NamePool.init(self.allocator);
```

Comment:
if this helper class can not operate with an allocator-free paradigm, then it should be marked as such in the main doc comment for `ReverseLookup`